Лабораторная 3

anpian13 opened the pull request.

anpian13 committed:
Лабораторная 3

dmitrylaabe commented on Лабораторная 3/main.py:
```diff
@@ -11,21 +19,47 @@ def __repr__(self):
+            raise TypeError("Количество страниц в книге должно быть типа int")
+        if pages <= 0:
+            raise ValueError("Количество страниц в книге должно быть больше 0")
         self.pages = pages
```
self.pages = pages - это ошибка, которая приводит к бесконечной рекурсии. Мы из setter вызываем этот же setter. Бесконечно

dmitrylaabe commented on Лабораторная 3/main.py:
```diff
@@ -11,21 +19,47 @@ def __repr__(self):
+            raise ValueError("Количество страниц в книге должно быть больше 0")
         self.pages = pages
 
     def __str__(self):
```
Нет необходимости переопределять __str__

dmitrylaabe commented on Лабораторная 3/main.py:
```diff
@@ -11,21 +19,47 @@ def __repr__(self):
+    def duration(self) -> int:
+        return self._duration
+
+    @duration.setter
```
А значение duration где задается в setter?

dmitrylaabe commented on Лабораторная 3/main.py:
```diff
@@ -11,21 +19,47 @@ def __repr__(self):
+        if duration <= 0:
+            raise ValueError("Продолжительность книги должна быть больше 0")
+
     def __str__(self):
```
Нет необходимости переопределять __str__

anpian13 committed:
Исправлены замечания

- Убраны переопределения str
- Значение duration задается в setter
- Исправлена ошибка с self.pages

anpian13 commented:
@dmitrylaabe Исправлено
